pagopa/io-app
# fix: [IA-650] Unpredictable status bar behavior

## Short description
The current codebase has several issues related to the status bar in both iOS and Android. If you navigate back and forth between screens that have different status bar settings (i.e. light vs. dark content) you will notice glitches and unpredictable behaviors. This is mainly due to:
1. the usage of both the declarative and imperative methods for setting the status bar;
2. the `Header` component (from the `native-base` library) that wraps the `StatusBar` component without taking into consideration the focus state.

The latter in particular has [several implications](https://reactnavigation.org/docs/4.x/status-bar#tabs-and-drawer) when used in a tab navigator.

## List of changes proposed in this pull request
- The `StatusBar` component has been removed from the `Header` in `native-base` using `patch-package`. This should be considered safe, as the patch is quite small and doesn't have any side effect. Even if we consider to upgrade the library, the `Header` component has been deprecated and [removed](https://docs.nativebase.io/next/migration/header) in newer releases of the library, so in that case we need to remove the `Header` component completely.

- The imperative method has been removed and should never be used again. By default each screen will inherit the dark content status bar (with white background) from the `RootContainer`. When you need to change the default appearance, a `FocusAwareStatusBar` component has been added (following the React Navigation [documentation](https://reactnavigation.org/docs/4.x/status-bar#tabs-and-drawer)), and this should always be used to change the status bar appearance inside a navigation scope. Outside of a navigation context, use the standard `StatusBar` component instead.

## How to test
Navigate throughout the app screens and verify that the status bar behaves correctly. If you find some unhandled screen  (where the status bar has a wrong appearance) let me know that I'll fix it.
